pixeltable/utils/media_store.py

```diff
@@ -7,114 +9,138 @@
+    def relocate_local_media_file(cls, src_path: Path, col: Column) -> str:
         """Relocate a local file to the MediaStore, and return its new URL"""
-        dest_path = MediaStore.prepare_media_path(tbl_id, col_id, tbl_version, ext=src_path.suffix)
+        dest_path = cls._prepare_media_path(col, ext=src_path.suffix)
         src_path.rename(dest_path)
         return urllib.parse.urljoin('file:', urllib.request.pathname2url(str(dest_path)))
 
     @classmethod
-    def save_media_file(cls, file_data: bytes, tbl_id: UUID, col_id: int, tbl_version: int) -> Path:
+    def save_media_object(cls, data: bytes | PIL.Image.Image, col: Column, format: Optional[str]) -> tuple[Path, str]:
+        """Save a media binary data to a file in the MediaStore
+        Returns:
+            dest_path: Path to the saved media file
+            url: URL of the saved media file
+        """
+        assert col.col_type.is_media_type(), f'MediaStore: request to store non media_type Column {col.name}'
+        dest_path = cls._prepare_media_path(col)
+        if isinstance(data, bytes):
+            dest_path = cls._save_binary_media_file(data, dest_path, format)
+        elif isinstance(data, PIL.Image.Image):
+            dest_path = cls._save_pil_image_file(data, dest_path, format)
+        else:
+            raise ValueError(f'Unsupported media object type: {type(data)}')
+        url = urllib.parse.urljoin('file:', urllib.request.pathname2url(str(dest_path)))
+        return dest_path, url
+
+    @classmethod
+    def _save_binary_media_file(cls, file_data: bytes, dest_path: Path, format: Optional[str]) -> Path:
         """Save a media binary data to a file in the MediaStore"""
         assert isinstance(file_data, bytes)
-        dest_path = cls.prepare_media_path(tbl_id, col_id, tbl_version)
         with open(dest_path, 'wb') as f:
             f.write(file_data)
             f.flush()  # Ensures Python buffers are written to OS
             os.fsync(f.fileno())  # Forces OS to write to physical storage
         return dest_path
 
     @classmethod
-    def save_image_file(cls, image: PIL.Image.Image, tbl_id: UUID, col_id: int, version: int) -> tuple[str, str]:
+    def _save_pil_image_file(cls, image: PIL.Image.Image, dest_path: Path, format: Optional[str]) -> Path:
         """Save a PIL Image to a file in the MediaStore"""
-        dest_path = str(MediaStore.prepare_media_path(tbl_id, col_id, version))
-        assert isinstance(image, PIL.Image.Image)
-        # Default to JPEG unless the image has a transparency layer (which isn't supported by JPEG).
-        # In that case, use WebP instead.
-        format = 'webp' if image.has_transparency_data else 'jpeg'
-        if not dest_path.endswith(f'.{format}'):
-            dest_path += f'.{format}'
-        image.save(dest_path, format=format)
-        url = urllib.parse.urljoin('file:', urllib.request.pathname2url(dest_path))
-        return dest_path, url
+        t_path = str(dest_path)
```

Comment:
This may be cleaner than using `t_path`:

```python
if dest_path.suffix != f'.{format}':
    dest_path = dest_path.with_name(f'{dest_path.name}.{format}')
```